src/Solucion_JavierRamiro/src/programacion/empresaprogramacion/Empresa.java

```diff
@@ -0,0 +1,132 @@
+package programacion.empresaprogramacion;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class Empresa {
+    private List<Empleado> empleados;
+
+    public Empresa(String nombre) {
+        this.empleados = new ArrayList<>();
+    }
+
+    public void addEmpleado(Empleado empleado) {
+        empleados.add(empleado);
+    }
+
+    public void toStringEmpleados() {
+        for (Empleado empleado : empleados) {
+            System.out.println(empleado);
+        }
+    }
+
+    public void toStringProgramadores() {
+        for (Empleado empleado : empleados) {
+            if (empleado.getTipo() == TipoEmpleado.PROGRAMADOR) {
+                System.out.println(empleado);
+            }
+        }
+    }
+
+    public void toStringOrdenSueldo() {
+        Collections.sort(empleados, new Comparator<Empleado>() {
+            @Override
+            public int compare(Empleado emp1, Empleado emp2) {
+                return Double.compare(emp2.getSueldo(), emp1.getSueldo());
+            }
+        });
+
+        for (Empleado empleado : empleados) {
+            System.out.println(empleado);
+        }
+    }
+
+    public void toStringOrdenNombre() {
+        Collections.sort(empleados, new Comparator<Empleado>() {
+            @Override
+            public int compare(Empleado emp1, Empleado emp2) {
+                return emp1.getNombre().compareTo(emp2.getNombre());
+            }
+        });
+
+        for (Empleado empleado : empleados) {
+            System.out.println(empleado);
+        }
+    }
+
+    public void cargarDesdeCSV(String archivo) {
+        try (BufferedReader br = new BufferedReader(new FileReader(archivo))) {
```

Comment:
br no es un nombre que indique lo que representa esa referencia. No abreviéis!
```suggestion
        try (BufferedReader lectorConBuffer = new BufferedReader(new FileReader(archivo))) {
```